### PR TITLE
refactor!: remove task specific data

### PR DESCRIPTION
**Pull request by mblottiere**
## Description

Remove task specific data.

TODO:
- [x] migration to drop data column
- [x] set logsPermissions
- [x] adapt e2e tests

## Checklist

- [x] [changelog](../CHANGELOG.md) was updated with notable changes
- [ ] documentation was updated

### CODE REVIEW

File: lib/service/computetask.go [hunk 1]
```diff
@@ -1001,6 +851,44 @@ func (s *ComputeTaskService) getTaskWorker(input *asset.NewComputeTask, algo *as
...
+		if !ok {
+			return nil, orcerrors.NewInvalidAsset(fmt.Sprintf("unknown task input: this identifier was not declared in the Algo: %q", identifier))
+		}
+
+		if algoInput.Kind == asset.AssetKind_ASSET_DATA_MANAGER {
+			dmKey := taskInput.GetAssetKey()
+			if dmKey == "" {
+				return nil, orcerrors.NewInvalidAsset(fmt.Sprintf("invalid task input %q: openers must be referenced using an asset key", taskInput.Identifier))
+			}
+			datamanager, err := s.getCachedDataManager(taskInput.GetAssetKey())
```
Comment (thibowk) [inline lib/service/computetask.go:871]:
```suggestion
			datamanager, err := s.getCachedDataManager(dmKey)
```

**Review by thibowk — APPROVED** [review]
Really nice to see this happening 🥳  !

**Comment by AurelienGasser** [comment]
Please run the e2e tests for significant changes

/e2e

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: lib/service/computetask.go [hunk 2]
```diff
@@ -1001,6 +851,44 @@ func (s *ComputeTaskService) getTaskWorker(input *asset.NewComputeTask, algo *as
...
+
+	// Fallback on parent union
+	logsPermission, err := s.GetPermissionService().CreatePermission(owner, &asset.NewPermissions{Public: false})
+	if err != nil {
+		return nil, err
+	}
+	for _, p := range parentTasks {
+		logsPermission = s.GetPermissionService().UnionPermission(p.LogsPermission, logsPermission)
+	}
+
```
Comment (AurelienGasser) [inline lib/service/computetask.go:888]: Return an error if there's no parents? There's nothing stopping a user from creating a task which has no parents and no data manager.

**Comment by AurelienGasser** [comment]
Please note that the migration in this PR conflicts with https://github.com/Substra/orchestrator/pull/48

File: lib/service/computetask.go (hunk 2 above)
Comment (Esadruhn) [reply lib/service/computetask.go:888]: please do not return an error, we are doing the PRs on the SDK to stop filling the parent_task_keys and data manager fields

**Comment by Esadruhn** [comment]
Companion PRs that need this branch to work, as they use the `/task` endpoint and do not fill the extra `Data` fields (setting `category` as unknown for now in the backend until https://github.com/Substra/orchestrator/pull/48 is merged):

https://github.com/Substra/substra/pull/299
https://github.com/Substra/substra-backend/pull/513
https://github.com/Substra/substrafl/pull/31
https://github.com/Substra/substra-tests/pull/221
https://github.com/Substra/substra-documentation/pull/197

File: lib/service/computetask.go (hunk 2 above)
Comment (mblottiere) [reply lib/service/computetask.go:888]: There's no issue if there's not parents, the permission will only allow the task owner to access the logs, which seems a sensible behavior.

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Pull request merged**